From 2898f8aa4e15cee474f14a2f1c1396127784eb28 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Fri, 27 Feb 2015 16:07:13 +0800
Subject: [PATCH] MLK-10258-1 IMX7D: EPXP: add dts support for the new PXP
 module on 7D board

Add the following dts support for the ehanced PXP
module on 7D board:
1. Add the new interrupt line introduced by EPXP.
   This interrupt is dedicated for the new working
   mode on EPXP.
2. Enable EPXP on 12x12 7D validation board by default.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts | 4 ++++
 arch/arm/boot/dts/imx7d.dtsi           | 5 ++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 7e61159..1eb3fc5 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -516,6 +516,10 @@
 	status = "okay";
 };
 
+&epxp {
+	status = "okay";
+};
+
 &i2c1 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default";
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index e581de0..c9a1463 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -695,8 +695,11 @@
 
 			epxp: epxp@30700000 {
 				compatible = "fsl,imx7d-pxp-dma";
-				interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 46 IRQ_TYPE_LEVEL_HIGH>;
 			        reg = <0x30700000 0x10000>;
+				clocks = <&clks IMX7D_CLK_DUMMY>;
+				clock-names = "pxp-axi";
 				status = "disabled";
 			};
 
-- 
1.8.0

